src/toast/Toast.tsx

```diff
@@ -0,0 +1,153 @@
+        //Default display time for toast is 3 sec
+        //Default type is SUCCESS
+        //Default position is top-right
+        const hideTime = (options.hideAfter === undefined ? 3 : options.hideAfter) * 1000;
+        const toastType = options.type === undefined ? MessageType.SUCCESS : options.type;
+        const pos = options.position === undefined ? Direction.TOP_RIGHT : options.position;
+
+        //As this is a simple toast, allowing one toast to be shown at a time
+        if (this.state.isToastRendering === false) {
+            //create styles based on the position given
+            let place = (pos || Direction.TOP_RIGHT).includes(Direction.BOTTOM)
+                ? Direction.CAPS_BOTTOM
+                : Direction.CAPS_TOP;
+            const marginTypeForTopOrBottom = `margin${place}`;
+            place = (pos || Direction.TOP_RIGHT).includes(Direction.LEFT) ? Direction.CAPS_LEFT : Direction.CAPS_RIGHT;
+            const marginTypeForLeftOrRight = `margin${place}`;
+            place = (pos || Direction.TOP_RIGHT).includes(Direction.LEFT) ? Direction.LEFT : Direction.RIGHT;
+            const isLeftOrRight = `${place}`;
+            place = (pos || Direction.TOP_RIGHT).includes(Direction.BOTTOM) ? Direction.BOTTOM : Direction.TOP;
+            const isBottomOrTop = `${place}`;
+
+            //set the state to show the toast and setTimeout to hide the toast
+            this.setState(
+                {
+                    isToastRendering: true,
+                    msg: message,
+                    type: toastType,
+                    position: pos,
+                    hideAfter: hideTime,
+                    style: {
+                        opacity: 1,
+                        [marginTypeForTopOrBottom]: MARGINS.POS_MARGIN,
+                        [isLeftOrRight]: 0,
+                        [marginTypeForLeftOrRight]: MARGINS.POS_MARGIN,
+                        [isBottomOrTop]: 0,
+                    },
+                },
+                () => {
+                    setTimeout(
+                        () =>
+                            this.setState({
+                                isToastRendering: false,
+                                msg: "",
+                                type: MessageType.SUCCESS,
+                                position: Direction.TOP_RIGHT,
+                                hideAfter: 3,
+                                style: {
+                                    opacity: 0,
+                                    [marginTypeForTopOrBottom]: MARGINS.NEG_MARGIN,
+                                    [marginTypeForLeftOrRight]: MARGINS.NEG_MARGIN,
+                                },
+                            }),
+                        hideTime,
+                    );
+                },
+            );
+        }
+    };
+    render() {
+        return (
+            <div className={`toast ${this.state.type}`} style={this.state.style}>
```

Comment:
is 'toast' here is classname ? if yes then the code should be {` ${"toast"} ${ this.state.type} `}

Reply:
Updated code